internal/server/statvar_group.go

```diff
@@ -138,3 +141,104 @@ func (s *Server) GetStatVarGroup(
 	}
 	return svgResp, nil
 }
+
+// GetStatVarGroupNode implements API for Mixer.GetStatVarGroupNodeRequest.
+func (s *Server) GetStatVarGroupNode(
+	ctx context.Context, in *pb.GetStatVarGroupNodeRequest) (
+	*pb.StatVarGroupNode, error) {
+	places := in.GetPlaces()
+	svg := in.GetStatVarGroup()
+
+	if svg == "" {
+		return nil, status.Errorf(
+			codes.InvalidArgument, "Missing required argument: stat_var_group")
+	}
+
+	triples, err := readTriples(ctx, s.store, buildTriplesKey([]string{svg}))
+	if err != nil {
+		return nil, err
+	}
+
+	result := &pb.StatVarGroupNode{}
+	childStatVars := []string{}
+
+	// Go through triples and populate result fields.
+	for _, t := range triples[svg].Triples {
+		if t.SubjectID == svg {
+			// SVG is subject
+			if t.Predicate == "specializationOf" {
+				// Parent SVG
+				result.ParentStatVarGroups = append(result.ParentStatVarGroups, t.ObjectID)
+			} else if t.Predicate == "name" {
+				result.AbsoluteName = t.ObjectValue
+			}
+		} else {
+			// SVG is object
+			if t.Predicate == "specializationOf" {
+				// Children SVG
+				result.ChildStatVarGroups = append(result.ChildStatVarGroups,
+					&pb.StatVarGroupNode_ChildSVG{
+						Id:          t.SubjectID,
+						DisplayName: t.SubjectName,
+					})
+			} else if t.Predicate == "memberOf" {
+				// Children SV
+				result.ChildStatVars = append(result.ChildStatVars,
+					&pb.StatVarGroupNode_ChildSV{
+						Id: t.SubjectID,
+					})
+				childStatVars = append(childStatVars, t.SubjectID)
+			}
+		}
+	}
+	// TODO(shifucun): Filter sv and svg given place information
+	log.Println(places)
+
+	// Fetch the child stat var triples and construct the names.
+	if len(childStatVars) > 0 {
+		statVarNames, err := fetchStatVarNames(ctx, s, childStatVars)
+		if err != nil {
+			return nil, err
+		}
+		for _, c := range result.ChildStatVars {
+			c.DisplayName = statVarNames[c.Id]
+		}
+	}
+	return result, nil
+}
+
+func fetchStatVarNames(ctx context.Context, s *Server, statVars []string) (
```

Comment:
Should we compute the names and put it in the "name" property of StatVar?

Reply:
That would be nice!